deployer/src/infra/csi_node.rs

```diff
@@ -0,0 +1,181 @@
+use super::*;
+use tokio::{
+    net::UnixStream,
+    time::{sleep, Duration},
+};
+use tonic::transport::{Endpoint, Uri};
+use tower::service_fn;
+
+use crate::IoEngine;
+use rpc::csi::{identity_client::IdentityClient, GetPluginInfoRequest};
+
+const CSI_NODE: &str = "csi-node";
+
+#[async_trait]
+impl ComponentAction for CsiNode {
+    fn configure(&self, options: &StartOptions, mut cfg: Builder) -> Result<Builder, Error> {
+        Ok(if !options.csi_node {
+            cfg
+        } else {
+            if options.build {
+                std::process::Command::new("cargo")
+                    .args(&["build", "-p", "csi-driver", "--bin", CSI_NODE])
+                    .status()?;
+            }
+
+            let local_nodes = if options.local_nodes {
+                options.io_engines
+            } else {
+                0
+            };
+
+            for i in 0 .. local_nodes {
+                cfg = CsiNode::with_local_node(i, options, cfg);
+            }
+
+            for i in 0 .. options.app_nodes() {
+                cfg = CsiNode::with_app_node(i, cfg);
+            }
+            cfg
+        })
+    }
+    async fn start(&self, options: &StartOptions, cfg: &ComposeTest) -> Result<(), Error> {
+        if options.csi_node {
+            let local_nodes = if options.local_nodes {
+                options.io_engines
+            } else {
+                0
+            };
+
+            for i in 0 .. local_nodes {
+                let container_name = Self::local_container_name(&IoEngine::name(i, options));
+                cfg.start(&container_name).await?;
+            }
+
+            for i in 0 .. options.app_nodes() {
+                cfg.start(&Self::container_name(i)).await?;
+            }
+        }
+        Ok(())
+    }
+
+    async fn wait_on(&self, options: &StartOptions, _cfg: &ComposeTest) -> Result<(), Error> {
+        if !options.csi_node {
+            return Ok(());
+        }
+
+        let local_nodes = if options.local_nodes {
+            options.io_engines
+        } else {
+            0
+        };
+
+        for i in 0 .. local_nodes {
+            CsiNode::wait_local_node(i, options).await?;
+        }
+
+        for i in 0 .. options.app_nodes() {
+            CsiNode::wait_app_node(i).await?;
+        }
+
+        Ok(())
+    }
+}
+
+impl CsiNode {
+    fn name(i: u32) -> String {
+        format!("app-node-{}", i + 1)
```

Comment:
Should this be "csi-node-{}"?

Reply:
I left it as app-node-{} as this is "simulating" an application node (which is where the csi-node runs)